src/routers/alpha-router/functions/best-swap-route.ts

```diff
@@ -429,18 +480,31 @@ export function getBestSwapRouteBy(
         (b) =>
           `${b.percent}% ${routeToString(b.route)} ${b.gasCostInUSD.toExact()}`
       ),
+      flatL1GasCostUSD: gasCostL1USD.toExact(),
     },
     'USD gas estimates of best route'
   );
 
   const estimatedGasUsedQuoteToken = sumFn(
     _.map(bestSwap, (routeWithValidQuote) => routeWithValidQuote.gasCostInToken)
-  );
+  ).add(gasCostL1QuoteToken);
 
-  const quote = sumFn(
+  let quote = sumFn(
     _.map(bestSwap, (routeWithValidQuote) => routeWithValidQuote.quote)
   );
 
+  // Adjust the quoteGasAdjusted for the l1 fee
+  const tradeType = bestSwap[0]?.tradeType;
+  if (tradeType == TradeType.EXACT_INPUT) {
+    const quoteGasAdjustedForL1 = quote.subtract(
+      gasCostsL1.gasCostL1QuoteToken
+    );
+    quoteGasAdjusted = quoteGasAdjustedForL1;
+  } else {
+    const quoteGasAdjustedForL1 = quote.add(gasCostsL1.gasCostL1QuoteToken);
+    quoteGasAdjusted = quoteGasAdjustedForL1;
+  }
+
```

Comment:
So here we are taking the raw quote for each route (not the gas adjusted quote for each route) and adjusting that with the L1 fee.

Don't we need to adjust `quoteGasAdjusted` so that we also include the regular L2 gas costs as well in our total?